handler/core/client/client.go

```diff
@@ -13,8 +13,8 @@ import (
 )
 
 type ClientCredentialsGrantHandler struct {
-	// Enigma is the algorithm responsible for creating a validatable, opaque string.
-	Enigma enigma.Enigma
+	// AccessTokenStrategy is the algorithm responsible for creating a validatable token.
+	AccessTokenStrategy core.AccessTokenStrategy
```

Comment:
instead of relying directly on the enigma interface, we now rely on `AccessTokenStrategy.GenerateAccessToken` which is responsible for generating the access token. So we can do something like: 

```go
handler := ClientCredentialsGrantHandler{
    AccessTokenStrategy: &JWTStrategy{},
}
```